src/main/java/frc/robot/subsystems/lights/Lights.java

```diff
@@ -0,0 +1,80 @@
+package frc.robot.subsystems.lights;
+
+import com.ctre.phoenix.led.CANdle;
+import com.ctre.phoenix.led.CANdle.LEDStripType;
+import com.ctre.phoenix.led.CANdleConfiguration;
+import com.ctre.phoenix.led.RainbowAnimation;
+import com.ctre.phoenix.led.StrobeAnimation;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.RobotMap;
+import java.util.TreeMap;
+
+public class Lights {
+  /**Tree map of light code enums and integers that 
+     * represent the R, G, B values of each light code enum */
+  private TreeMap<LightCode, Integer[]> lightOptionsMap;
+  private CANdle candle = new CANdle(RobotMap.CANDLE_CAN_ID);
+  private CANdleConfiguration config = new CANdleConfiguration();
+  private LightCode currentLightStatus = LightCode.OFF;
+
+  public enum LightCode{
+    NOTELESS, //red
+    INTAKING, //blue
+    READY_TO_SHOOT, //green
+    ALIGNING_TO_TAG, //orange
+    HOLDING_NOTE, //yellow
+    OFF, //no lights are on
+    PARTY_MODE;
+  }
+
+  public Lights() {
+    config.stripType = LEDStripType.RGB;
+    config.brightnessScalar = 1.0;
+    candle.configAllSettings(config);
+
+    
+    lightOptionsMap = new TreeMap<LightCode, Integer[]>();
+    lightOptionsMap.put(LightCode.NOTELESS, new Integer[] {255, 0, 0});
+    lightOptionsMap.put(LightCode.INTAKING, new Integer[] {0, 0, 255});
+    lightOptionsMap.put(LightCode.READY_TO_SHOOT, new Integer[] {0, 255, 0});
+    lightOptionsMap.put(LightCode.OFF, new Integer[] {0, 0, 0});
+    lightOptionsMap.put(LightCode.ALIGNING_TO_TAG, new Integer[] {255, 165, 0});
+    lightOptionsMap.put(LightCode.HOLDING_NOTE, new Integer[] {255, 255, 102});
+  }
+
+  public void toggleCode(LightCode light) {
+    if (currentLightStatus == light) {
+      currentLightStatus = LightCode.OFF;
+    } else {
+      currentLightStatus = light;
+    }
+    setLEDs();
+  }
+
+  private void setLEDs() {
+    if (currentLightStatus == LightCode.PARTY_MODE) {
+      setPartyMode();
+    } else if (currentLightStatus == LightCode.NO_TAG) {
```

Comment:
you deleted NO_TAG from the treemap, which was good. you also deleted it as a LightCode but you still need NO_TAG as an enum option (that way you can set it as a desired color and stuff). so i don't think this will build